javascriptlib/src/mill/javascriptlib/TypeScriptModule.scala

```diff
@@ -56,28 +68,40 @@ trait TypeScriptModule extends Module {
     (PathRef(javascriptOut), PathRef(declarationsOut))
   }
 
-  def mainFileName = Task { s"${millSourcePath.last}.js" }
+  def mainFileName: Target[String] = Task { s"${millSourcePath.last}.js" }
 
-  def prepareRun = Task.Anon {
+  def mainFilePath: Target[Path] = Task { compile()._1.path / mainFileName() }
+
+  def mkENV =
+    Task.Anon { Map("NODE_PATH" -> Seq(".", compile()._1.path, npmInstall().path).mkString(":")) }
+
+  def prepareRun: Task[(Path, Map[String, String])] = Task.Anon {
     val upstream = Task.traverse(moduleDeps)(_.compile)().zip(moduleDeps)
-    for (((jsDir, tTsDir), mod) <- upstream) {
+    for (((jsDir, _), mod) <- upstream) {
       os.copy(jsDir.path, Task.dest / mod.millSourcePath.subRelativeTo(Task.workspace))
     }
-    val mainFile = compile()._1.path / mainFileName()
-    val env = Map("NODE_PATH" -> Seq(".", compile()._1.path, npmInstall().path).mkString(":"))
-    (mainFile, env)
+    (mainFilePath(), mkENV())
   }
 
-  def run(args: mill.define.Args) = Task.Command {
+  def argsOrder: Task[(Boolean, Seq[String])] = Task { (false, Seq("")) }
```

Comment:
What's this task about?

Reply:
to include computed arguments during run, the boolean is to specify if/not to include before/after user defined arguments

Reply:
Why does whether we include them before or after matter?

Reply:
The task type signature looks bery weird. If we really need to do this, can we break it into teo separate tasks each one a `Seq[String]`, one that gets inserted before user args and one after?

Reply:
I did instead: ` Task[(Seq[String], Seq[String])]`